### ISSUE TITLE
Implement our own backend.

### ISSUE THREAD

**Issue by hbhungg** [hbhungg/smalltensor#10]
C or Rust, as an exercise.
Maybe for GPU as well?

**Comment by AmitPress** [comment]
How does tinygrad do it? I think they use pycuda and pyopencl...

**Comment by hbhungg** [comment]
Afaik, tinygrad compiles their computational graphs down to C code, then use pyopencl (or other libs depending on your choices). I have not been following tinygrad lately so it might not be correct.

Anyways, there are no concrete plans for these, feels free to propose. Ideally would be something simple since this is just a learning projects.

**Comment by AmitPress** [comment]
> Anyways, there are no concrete plans for these, feels free to propose. Ideally would be something simple since this is just a learning projects.

Yeah, I am also trying to understand tinygrad. Its good to learn by doing.
I am more curious about the backend part like how it runs on gpu or other accelerators.
Thinking of forking this repo. If you don't mind, I would like to connect with you.

**Comment by hbhungg** [comment]
@AmitPress feels free to fork and work on it. What contact method would you prefer, Discord, email?

**Comment by AmitPress** [comment]
@hbhungg discord will be fine...

**Comment by hbhungg** [comment]
@AmitPress sure, whats your discord?

**Comment by AmitPress** [comment]
@hbhungg ammunition#7150
